# Data Preparation

Okay, with all our data in json files, we are ready for our next step, data preparation.

The data are not be in a format that we like or are not usable, so we have to prepare the data for our use and extract the relevant information. 



In [21]:
import pandas as pd
import json
import random

## Formatting data and convert into dataframe

First, we import in the data from our movie detail json file.

In [22]:
def read_json(filename: str) -> dict:
  with open(filename, 'r') as f:
    return json.load(f)

### 1. Cleaning Data

In our code below, we clean up the data, placing different variables in separate lists.

When cleaning up our data, we only extract out the variables that are relevant.

In [23]:
def cleaning_data (movie_detail):
    clean_data = {'id':[],'runtime':[],'genres':[],'keywords':[],'production_companies':[],'age_rating':[],
                  'budget':[]}
    i=0
    for movie in movie_detail:
        i+=1
        print("Cleaning", i , "-th movie")
        st = False
        
        if movie != None:
            for certification in movie["release_dates"]['results']:
                
                if (certification["iso_3166_1"] == 'US'):
                    st = True
                    clean_data['age_rating'].append(certification['release_dates'][0]["certification"])
                    break
            
            if st==False:
                # This is when the data does not have US rating, so we have to ignore the data
                continue

            # id
            clean_data['id'].append(movie['id'])

            # runtime
            clean_data['runtime'].append(movie['runtime'])

            #genres
            genres_list = []
            for genre in movie['genres']:
                genres_list.append(genre['name'])
            clean_data['genres'].append(genres_list)
            
            #Keywords
            keywords_list = []
            for keywords in movie['keywords']['keywords']:
                keywords_list.append(keywords['name'])
            clean_data['keywords'].append(keywords_list)
            production_companies = []
            for production in movie['production_companies']:
                production_companies.append(production['name'])
            clean_data['production_companies'].append(production_companies)

            #budget
            clean_data['budget'].append(movie['budget'])

            
    return clean_data

In [24]:
movie_detail = read_json("movie_detail.json")
print("The length of movie_detail is ", len(movie_detail))

The length of movie_detail is  10000


In [25]:
clean_data = cleaning_data(movie_detail)

Cleaning 1 -th movie
Cleaning 2 -th movie
Cleaning 3 -th movie
Cleaning 4 -th movie
Cleaning 5 -th movie
Cleaning 6 -th movie
Cleaning 7 -th movie
Cleaning 8 -th movie
Cleaning 9 -th movie
Cleaning 10 -th movie
Cleaning 11 -th movie
Cleaning 12 -th movie
Cleaning 13 -th movie
Cleaning 14 -th movie
Cleaning 15 -th movie
Cleaning 16 -th movie
Cleaning 17 -th movie
Cleaning 18 -th movie
Cleaning 19 -th movie
Cleaning 20 -th movie
Cleaning 21 -th movie
Cleaning 22 -th movie
Cleaning 23 -th movie
Cleaning 24 -th movie
Cleaning 25 -th movie
Cleaning 26 -th movie
Cleaning 27 -th movie
Cleaning 28 -th movie
Cleaning 29 -th movie
Cleaning 30 -th movie
Cleaning 31 -th movie
Cleaning 32 -th movie
Cleaning 33 -th movie
Cleaning 34 -th movie
Cleaning 35 -th movie
Cleaning 36 -th movie
Cleaning 37 -th movie
Cleaning 38 -th movie
Cleaning 39 -th movie
Cleaning 40 -th movie
Cleaning 41 -th movie
Cleaning 42 -th movie
Cleaning 43 -th movie
Cleaning 44 -th movie
Cleaning 45 -th movie
Cleaning 46 -th mov

Cleaning 718 -th movie
Cleaning 719 -th movie
Cleaning 720 -th movie
Cleaning 721 -th movie
Cleaning 722 -th movie
Cleaning 723 -th movie
Cleaning 724 -th movie
Cleaning 725 -th movie
Cleaning 726 -th movie
Cleaning 727 -th movie
Cleaning 728 -th movie
Cleaning 729 -th movie
Cleaning 730 -th movie
Cleaning 731 -th movie
Cleaning 732 -th movie
Cleaning 733 -th movie
Cleaning 734 -th movie
Cleaning 735 -th movie
Cleaning 736 -th movie
Cleaning 737 -th movie
Cleaning 738 -th movie
Cleaning 739 -th movie
Cleaning 740 -th movie
Cleaning 741 -th movie
Cleaning 742 -th movie
Cleaning 743 -th movie
Cleaning 744 -th movie
Cleaning 745 -th movie
Cleaning 746 -th movie
Cleaning 747 -th movie
Cleaning 748 -th movie
Cleaning 749 -th movie
Cleaning 750 -th movie
Cleaning 751 -th movie
Cleaning 752 -th movie
Cleaning 753 -th movie
Cleaning 754 -th movie
Cleaning 755 -th movie
Cleaning 756 -th movie
Cleaning 757 -th movie
Cleaning 758 -th movie
Cleaning 759 -th movie
Cleaning 760 -th movie
Cleaning 76

Cleaning 1384 -th movie
Cleaning 1385 -th movie
Cleaning 1386 -th movie
Cleaning 1387 -th movie
Cleaning 1388 -th movie
Cleaning 1389 -th movie
Cleaning 1390 -th movie
Cleaning 1391 -th movie
Cleaning 1392 -th movie
Cleaning 1393 -th movie
Cleaning 1394 -th movie
Cleaning 1395 -th movie
Cleaning 1396 -th movie
Cleaning 1397 -th movie
Cleaning 1398 -th movie
Cleaning 1399 -th movie
Cleaning 1400 -th movie
Cleaning 1401 -th movie
Cleaning 1402 -th movie
Cleaning 1403 -th movie
Cleaning 1404 -th movie
Cleaning 1405 -th movie
Cleaning 1406 -th movie
Cleaning 1407 -th movie
Cleaning 1408 -th movie
Cleaning 1409 -th movie
Cleaning 1410 -th movie
Cleaning 1411 -th movie
Cleaning 1412 -th movie
Cleaning 1413 -th movie
Cleaning 1414 -th movie
Cleaning 1415 -th movie
Cleaning 1416 -th movie
Cleaning 1417 -th movie
Cleaning 1418 -th movie
Cleaning 1419 -th movie
Cleaning 1420 -th movie
Cleaning 1421 -th movie
Cleaning 1422 -th movie
Cleaning 1423 -th movie
Cleaning 1424 -th movie
Cleaning 1425 -t

Cleaning 2051 -th movie
Cleaning 2052 -th movie
Cleaning 2053 -th movie
Cleaning 2054 -th movie
Cleaning 2055 -th movie
Cleaning 2056 -th movie
Cleaning 2057 -th movie
Cleaning 2058 -th movie
Cleaning 2059 -th movie
Cleaning 2060 -th movie
Cleaning 2061 -th movie
Cleaning 2062 -th movie
Cleaning 2063 -th movie
Cleaning 2064 -th movie
Cleaning 2065 -th movie
Cleaning 2066 -th movie
Cleaning 2067 -th movie
Cleaning 2068 -th movie
Cleaning 2069 -th movie
Cleaning 2070 -th movie
Cleaning 2071 -th movie
Cleaning 2072 -th movie
Cleaning 2073 -th movie
Cleaning 2074 -th movie
Cleaning 2075 -th movie
Cleaning 2076 -th movie
Cleaning 2077 -th movie
Cleaning 2078 -th movie
Cleaning 2079 -th movie
Cleaning 2080 -th movie
Cleaning 2081 -th movie
Cleaning 2082 -th movie
Cleaning 2083 -th movie
Cleaning 2084 -th movie
Cleaning 2085 -th movie
Cleaning 2086 -th movie
Cleaning 2087 -th movie
Cleaning 2088 -th movie
Cleaning 2089 -th movie
Cleaning 2090 -th movie
Cleaning 2091 -th movie
Cleaning 2092 -t

Cleaning 2806 -th movie
Cleaning 2807 -th movie
Cleaning 2808 -th movie
Cleaning 2809 -th movie
Cleaning 2810 -th movie
Cleaning 2811 -th movie
Cleaning 2812 -th movie
Cleaning 2813 -th movie
Cleaning 2814 -th movie
Cleaning 2815 -th movie
Cleaning 2816 -th movie
Cleaning 2817 -th movie
Cleaning 2818 -th movie
Cleaning 2819 -th movie
Cleaning 2820 -th movie
Cleaning 2821 -th movie
Cleaning 2822 -th movie
Cleaning 2823 -th movie
Cleaning 2824 -th movie
Cleaning 2825 -th movie
Cleaning 2826 -th movie
Cleaning 2827 -th movie
Cleaning 2828 -th movie
Cleaning 2829 -th movie
Cleaning 2830 -th movie
Cleaning 2831 -th movie
Cleaning 2832 -th movie
Cleaning 2833 -th movie
Cleaning 2834 -th movie
Cleaning 2835 -th movie
Cleaning 2836 -th movie
Cleaning 2837 -th movie
Cleaning 2838 -th movie
Cleaning 2839 -th movie
Cleaning 2840 -th movie
Cleaning 2841 -th movie
Cleaning 2842 -th movie
Cleaning 2843 -th movie
Cleaning 2844 -th movie
Cleaning 2845 -th movie
Cleaning 2846 -th movie
Cleaning 2847 -t

Cleaning 3432 -th movie
Cleaning 3433 -th movie
Cleaning 3434 -th movie
Cleaning 3435 -th movie
Cleaning 3436 -th movie
Cleaning 3437 -th movie
Cleaning 3438 -th movie
Cleaning 3439 -th movie
Cleaning 3440 -th movie
Cleaning 3441 -th movie
Cleaning 3442 -th movie
Cleaning 3443 -th movie
Cleaning 3444 -th movie
Cleaning 3445 -th movie
Cleaning 3446 -th movie
Cleaning 3447 -th movie
Cleaning 3448 -th movie
Cleaning 3449 -th movie
Cleaning 3450 -th movie
Cleaning 3451 -th movie
Cleaning 3452 -th movie
Cleaning 3453 -th movie
Cleaning 3454 -th movie
Cleaning 3455 -th movie
Cleaning 3456 -th movie
Cleaning 3457 -th movie
Cleaning 3458 -th movie
Cleaning 3459 -th movie
Cleaning 3460 -th movie
Cleaning 3461 -th movie
Cleaning 3462 -th movie
Cleaning 3463 -th movie
Cleaning 3464 -th movie
Cleaning 3465 -th movie
Cleaning 3466 -th movie
Cleaning 3467 -th movie
Cleaning 3468 -th movie
Cleaning 3469 -th movie
Cleaning 3470 -th movie
Cleaning 3471 -th movie
Cleaning 3472 -th movie
Cleaning 3473 -t

Cleaning 4217 -th movie
Cleaning 4218 -th movie
Cleaning 4219 -th movie
Cleaning 4220 -th movie
Cleaning 4221 -th movie
Cleaning 4222 -th movie
Cleaning 4223 -th movie
Cleaning 4224 -th movie
Cleaning 4225 -th movie
Cleaning 4226 -th movie
Cleaning 4227 -th movie
Cleaning 4228 -th movie
Cleaning 4229 -th movie
Cleaning 4230 -th movie
Cleaning 4231 -th movie
Cleaning 4232 -th movie
Cleaning 4233 -th movie
Cleaning 4234 -th movie
Cleaning 4235 -th movie
Cleaning 4236 -th movie
Cleaning 4237 -th movie
Cleaning 4238 -th movie
Cleaning 4239 -th movie
Cleaning 4240 -th movie
Cleaning 4241 -th movie
Cleaning 4242 -th movie
Cleaning 4243 -th movie
Cleaning 4244 -th movie
Cleaning 4245 -th movie
Cleaning 4246 -th movie
Cleaning 4247 -th movie
Cleaning 4248 -th movie
Cleaning 4249 -th movie
Cleaning 4250 -th movie
Cleaning 4251 -th movie
Cleaning 4252 -th movie
Cleaning 4253 -th movie
Cleaning 4254 -th movie
Cleaning 4255 -th movie
Cleaning 4256 -th movie
Cleaning 4257 -th movie
Cleaning 4258 -t

Cleaning 5089 -th movie
Cleaning 5090 -th movie
Cleaning 5091 -th movie
Cleaning 5092 -th movie
Cleaning 5093 -th movie
Cleaning 5094 -th movie
Cleaning 5095 -th movie
Cleaning 5096 -th movie
Cleaning 5097 -th movie
Cleaning 5098 -th movie
Cleaning 5099 -th movie
Cleaning 5100 -th movie
Cleaning 5101 -th movie
Cleaning 5102 -th movie
Cleaning 5103 -th movie
Cleaning 5104 -th movie
Cleaning 5105 -th movie
Cleaning 5106 -th movie
Cleaning 5107 -th movie
Cleaning 5108 -th movie
Cleaning 5109 -th movie
Cleaning 5110 -th movie
Cleaning 5111 -th movie
Cleaning 5112 -th movie
Cleaning 5113 -th movie
Cleaning 5114 -th movie
Cleaning 5115 -th movie
Cleaning 5116 -th movie
Cleaning 5117 -th movie
Cleaning 5118 -th movie
Cleaning 5119 -th movie
Cleaning 5120 -th movie
Cleaning 5121 -th movie
Cleaning 5122 -th movie
Cleaning 5123 -th movie
Cleaning 5124 -th movie
Cleaning 5125 -th movie
Cleaning 5126 -th movie
Cleaning 5127 -th movie
Cleaning 5128 -th movie
Cleaning 5129 -th movie
Cleaning 5130 -t

Cleaning 5883 -th movie
Cleaning 5884 -th movie
Cleaning 5885 -th movie
Cleaning 5886 -th movie
Cleaning 5887 -th movie
Cleaning 5888 -th movie
Cleaning 5889 -th movie
Cleaning 5890 -th movie
Cleaning 5891 -th movie
Cleaning 5892 -th movie
Cleaning 5893 -th movie
Cleaning 5894 -th movie
Cleaning 5895 -th movie
Cleaning 5896 -th movie
Cleaning 5897 -th movie
Cleaning 5898 -th movie
Cleaning 5899 -th movie
Cleaning 5900 -th movie
Cleaning 5901 -th movie
Cleaning 5902 -th movie
Cleaning 5903 -th movie
Cleaning 5904 -th movie
Cleaning 5905 -th movie
Cleaning 5906 -th movie
Cleaning 5907 -th movie
Cleaning 5908 -th movie
Cleaning 5909 -th movie
Cleaning 5910 -th movie
Cleaning 5911 -th movie
Cleaning 5912 -th movie
Cleaning 5913 -th movie
Cleaning 5914 -th movie
Cleaning 5915 -th movie
Cleaning 5916 -th movie
Cleaning 5917 -th movie
Cleaning 5918 -th movie
Cleaning 5919 -th movie
Cleaning 5920 -th movie
Cleaning 5921 -th movie
Cleaning 5922 -th movie
Cleaning 5923 -th movie
Cleaning 5924 -t

Cleaning 6717 -th movie
Cleaning 6718 -th movie
Cleaning 6719 -th movie
Cleaning 6720 -th movie
Cleaning 6721 -th movie
Cleaning 6722 -th movie
Cleaning 6723 -th movie
Cleaning 6724 -th movie
Cleaning 6725 -th movie
Cleaning 6726 -th movie
Cleaning 6727 -th movie
Cleaning 6728 -th movie
Cleaning 6729 -th movie
Cleaning 6730 -th movie
Cleaning 6731 -th movie
Cleaning 6732 -th movie
Cleaning 6733 -th movie
Cleaning 6734 -th movie
Cleaning 6735 -th movie
Cleaning 6736 -th movie
Cleaning 6737 -th movie
Cleaning 6738 -th movie
Cleaning 6739 -th movie
Cleaning 6740 -th movie
Cleaning 6741 -th movie
Cleaning 6742 -th movie
Cleaning 6743 -th movie
Cleaning 6744 -th movie
Cleaning 6745 -th movie
Cleaning 6746 -th movie
Cleaning 6747 -th movie
Cleaning 6748 -th movie
Cleaning 6749 -th movie
Cleaning 6750 -th movie
Cleaning 6751 -th movie
Cleaning 6752 -th movie
Cleaning 6753 -th movie
Cleaning 6754 -th movie
Cleaning 6755 -th movie
Cleaning 6756 -th movie
Cleaning 6757 -th movie
Cleaning 6758 -t

Cleaning 7550 -th movie
Cleaning 7551 -th movie
Cleaning 7552 -th movie
Cleaning 7553 -th movie
Cleaning 7554 -th movie
Cleaning 7555 -th movie
Cleaning 7556 -th movie
Cleaning 7557 -th movie
Cleaning 7558 -th movie
Cleaning 7559 -th movie
Cleaning 7560 -th movie
Cleaning 7561 -th movie
Cleaning 7562 -th movie
Cleaning 7563 -th movie
Cleaning 7564 -th movie
Cleaning 7565 -th movie
Cleaning 7566 -th movie
Cleaning 7567 -th movie
Cleaning 7568 -th movie
Cleaning 7569 -th movie
Cleaning 7570 -th movie
Cleaning 7571 -th movie
Cleaning 7572 -th movie
Cleaning 7573 -th movie
Cleaning 7574 -th movie
Cleaning 7575 -th movie
Cleaning 7576 -th movie
Cleaning 7577 -th movie
Cleaning 7578 -th movie
Cleaning 7579 -th movie
Cleaning 7580 -th movie
Cleaning 7581 -th movie
Cleaning 7582 -th movie
Cleaning 7583 -th movie
Cleaning 7584 -th movie
Cleaning 7585 -th movie
Cleaning 7586 -th movie
Cleaning 7587 -th movie
Cleaning 7588 -th movie
Cleaning 7589 -th movie
Cleaning 7590 -th movie
Cleaning 7591 -t

Cleaning 8383 -th movie
Cleaning 8384 -th movie
Cleaning 8385 -th movie
Cleaning 8386 -th movie
Cleaning 8387 -th movie
Cleaning 8388 -th movie
Cleaning 8389 -th movie
Cleaning 8390 -th movie
Cleaning 8391 -th movie
Cleaning 8392 -th movie
Cleaning 8393 -th movie
Cleaning 8394 -th movie
Cleaning 8395 -th movie
Cleaning 8396 -th movie
Cleaning 8397 -th movie
Cleaning 8398 -th movie
Cleaning 8399 -th movie
Cleaning 8400 -th movie
Cleaning 8401 -th movie
Cleaning 8402 -th movie
Cleaning 8403 -th movie
Cleaning 8404 -th movie
Cleaning 8405 -th movie
Cleaning 8406 -th movie
Cleaning 8407 -th movie
Cleaning 8408 -th movie
Cleaning 8409 -th movie
Cleaning 8410 -th movie
Cleaning 8411 -th movie
Cleaning 8412 -th movie
Cleaning 8413 -th movie
Cleaning 8414 -th movie
Cleaning 8415 -th movie
Cleaning 8416 -th movie
Cleaning 8417 -th movie
Cleaning 8418 -th movie
Cleaning 8419 -th movie
Cleaning 8420 -th movie
Cleaning 8421 -th movie
Cleaning 8422 -th movie
Cleaning 8423 -th movie
Cleaning 8424 -t

Cleaning 9216 -th movie
Cleaning 9217 -th movie
Cleaning 9218 -th movie
Cleaning 9219 -th movie
Cleaning 9220 -th movie
Cleaning 9221 -th movie
Cleaning 9222 -th movie
Cleaning 9223 -th movie
Cleaning 9224 -th movie
Cleaning 9225 -th movie
Cleaning 9226 -th movie
Cleaning 9227 -th movie
Cleaning 9228 -th movie
Cleaning 9229 -th movie
Cleaning 9230 -th movie
Cleaning 9231 -th movie
Cleaning 9232 -th movie
Cleaning 9233 -th movie
Cleaning 9234 -th movie
Cleaning 9235 -th movie
Cleaning 9236 -th movie
Cleaning 9237 -th movie
Cleaning 9238 -th movie
Cleaning 9239 -th movie
Cleaning 9240 -th movie
Cleaning 9241 -th movie
Cleaning 9242 -th movie
Cleaning 9243 -th movie
Cleaning 9244 -th movie
Cleaning 9245 -th movie
Cleaning 9246 -th movie
Cleaning 9247 -th movie
Cleaning 9248 -th movie
Cleaning 9249 -th movie
Cleaning 9250 -th movie
Cleaning 9251 -th movie
Cleaning 9252 -th movie
Cleaning 9253 -th movie
Cleaning 9254 -th movie
Cleaning 9255 -th movie
Cleaning 9256 -th movie
Cleaning 9257 -t

### 2. Converting into DataFrame

After cleaning our data, we create a DataFrame for the clean data.

This will allow us to use and manipulate the data as we wish.

In [26]:
df = pd.DataFrame(clean_data)
df['age_rating'].astype("category")
print("The shape of the dataFrame is" , df.shape)
df.head()

The shape of the dataFrame is (9038, 7)


id  runtime                                           genres  \
0  527774    107.0  [Animation, Adventure, Fantasy, Family, Action]   
1  587807    101.0   [Action, Comedy, Family, Animation, Adventure]   
2  484718    110.0                                         [Comedy]   
3  458576    104.0                     [Fantasy, Action, Adventure]   
4  464052    151.0                     [Fantasy, Action, Adventure]   

                                            keywords  \
0  [kung fu, trip, sword, warrior woman, computer...   
1  [computer animation, racial stereotype, live a...   
2  [africa, prince, heir to the throne, sequel, q...   
3  [female soldier, another dimension, bow and ar...   
4  [dc comics, superhero, based on comic, 1980s, ...   

                                production_companies age_rating     budget  
0  [Walt Disney Pictures, Walt Disney Animation S...         PG          0  
1  [Warner Animation Group, Warner Bros. Pictures...         PG   50000000  
2  [Paramount, New Republic Pictures, Eddie Murph...      PG-13   60000000  
3  [CAPCOM, Constantin Film, Impact Pictures, Ten...      PG-13   60000000  
4  [DC Entertainment, Warner Bros. Pictures, The ...      PG-13  200000000

Extract data that is only in the 5 interest Age_rating (refer to the data exploration - there is NR)

In [27]:
df = df [ df['age_rating'].isin(['PG','PG-13','R','NC-17','G'])]
print("The shape of the dataFrame is" , df.shape)
df

The shape of the dataFrame is (6596, 7)


id  runtime                                           genres  \
0     527774    107.0  [Animation, Adventure, Fantasy, Family, Action]   
1     587807    101.0   [Action, Comedy, Family, Animation, Adventure]   
2     484718    110.0                                         [Comedy]   
3     458576    104.0                     [Fantasy, Action, Adventure]   
4     464052    151.0                     [Fantasy, Action, Adventure]   
...      ...      ...                                              ...   
9031  226140     90.0                                         [Horror]   
9032  590123     84.0                              [Mystery, TV Movie]   
9033    9559    138.0                         [Crime, Drama, Thriller]   
9035   29723    123.0                        [Action, Drama, Thriller]   
9036   69945     86.0                      [Drama, Thriller, TV Movie]   

                                               keywords  \
0     [kung fu, trip, sword, warrior woman, computer...   
1     [computer animation, racial stereotype, live a...   
2     [africa, prince, heir to the throne, sequel, q...   
3     [female soldier, another dimension, bow and ar...   
4     [dc comics, superhero, based on comic, 1980s, ...   
...                                                 ...   
9031                  [sequel, slasher, woman director]   
9032                                                 []   
9033  [texas, hostage, escaped convict, prison escap...   
9035  [stewardess, missile, pilot, hot air balloon, ...   
9036  [lolita, babysitter, seduction, high school, m...   

                                   production_companies age_rating     budget  
0     [Walt Disney Pictures, Walt Disney Animation S...         PG          0  
1     [Warner Animation Group, Warner Bros. Pictures...         PG   50000000  
2     [Paramount, New Republic Pictures, Eddie Murph...      PG-13   60000000  
3     [CAPCOM, Constantin Film, Impact Pictures, Ten...      PG-13   60000000  
4     [DC Entertainment, Warner Bros. Pictures, The ...      PG-13  200000000  
...                                                 ...        ...        ...  
9031                           [WWE Studios, Lionsgate]          R          0  
9032                               [Muse Entertainment]         PG          0  
9033       [Malpaso Productions, Warner Bros. Pictures]      PG-13   30000000  
9035                               [Universal Pictures]         PG   14000000  
9036  [Hearst Entertainment Productions, Richard May...      PG-13          0  

[6596 rows x 7 columns]

There are 10000 movie data initial, but after running to get the movie_Details, there are only 9038 data that is valid (have US age_rating and also not empty data) left. However some of the data contains age_rating NR (which is No Rating and we ignore this data as stated in Data Exploration), we left 6596 data

In [28]:
df.to_csv("movie_data.csv", header=True,index=False)

# Cleaning the Additional data due to the imbalance of data

Due to the imbalance of data mentioned in Data Exploration.ipynb and we have extract more data. Therefore this is the part where we clean the extension of the data

Here,we do the same procedures here

In [29]:
movie_details = read_json("movie_detail_NC17_addtional.json")
print("The length of additional movie_detail is ", len(movie_details))

The length of additional movie_detail is  373


Extend the previous data and clean again

In [30]:
movie_detail.extend(movie_details)

In [31]:
clean_data_NC17_addtional = cleaning_data(movie_detail)

Cleaning 1 -th movie
Cleaning 2 -th movie
Cleaning 3 -th movie
Cleaning 4 -th movie
Cleaning 5 -th movie
Cleaning 6 -th movie
Cleaning 7 -th movie
Cleaning 8 -th movie
Cleaning 9 -th movie
Cleaning 10 -th movie
Cleaning 11 -th movie
Cleaning 12 -th movie
Cleaning 13 -th movie
Cleaning 14 -th movie
Cleaning 15 -th movie
Cleaning 16 -th movie
Cleaning 17 -th movie
Cleaning 18 -th movie
Cleaning 19 -th movie
Cleaning 20 -th movie
Cleaning 21 -th movie
Cleaning 22 -th movie
Cleaning 23 -th movie
Cleaning 24 -th movie
Cleaning 25 -th movie
Cleaning 26 -th movie
Cleaning 27 -th movie
Cleaning 28 -th movie
Cleaning 29 -th movie
Cleaning 30 -th movie
Cleaning 31 -th movie
Cleaning 32 -th movie
Cleaning 33 -th movie
Cleaning 34 -th movie
Cleaning 35 -th movie
Cleaning 36 -th movie
Cleaning 37 -th movie
Cleaning 38 -th movie
Cleaning 39 -th movie
Cleaning 40 -th movie
Cleaning 41 -th movie
Cleaning 42 -th movie
Cleaning 43 -th movie
Cleaning 44 -th movie
Cleaning 45 -th movie
Cleaning 46 -th mov

Cleaning 701 -th movie
Cleaning 702 -th movie
Cleaning 703 -th movie
Cleaning 704 -th movie
Cleaning 705 -th movie
Cleaning 706 -th movie
Cleaning 707 -th movie
Cleaning 708 -th movie
Cleaning 709 -th movie
Cleaning 710 -th movie
Cleaning 711 -th movie
Cleaning 712 -th movie
Cleaning 713 -th movie
Cleaning 714 -th movie
Cleaning 715 -th movie
Cleaning 716 -th movie
Cleaning 717 -th movie
Cleaning 718 -th movie
Cleaning 719 -th movie
Cleaning 720 -th movie
Cleaning 721 -th movie
Cleaning 722 -th movie
Cleaning 723 -th movie
Cleaning 724 -th movie
Cleaning 725 -th movie
Cleaning 726 -th movie
Cleaning 727 -th movie
Cleaning 728 -th movie
Cleaning 729 -th movie
Cleaning 730 -th movie
Cleaning 731 -th movie
Cleaning 732 -th movie
Cleaning 733 -th movie
Cleaning 734 -th movie
Cleaning 735 -th movie
Cleaning 736 -th movie
Cleaning 737 -th movie
Cleaning 738 -th movie
Cleaning 739 -th movie
Cleaning 740 -th movie
Cleaning 741 -th movie
Cleaning 742 -th movie
Cleaning 743 -th movie
Cleaning 74

Cleaning 1534 -th movie
Cleaning 1535 -th movie
Cleaning 1536 -th movie
Cleaning 1537 -th movie
Cleaning 1538 -th movie
Cleaning 1539 -th movie
Cleaning 1540 -th movie
Cleaning 1541 -th movie
Cleaning 1542 -th movie
Cleaning 1543 -th movie
Cleaning 1544 -th movie
Cleaning 1545 -th movie
Cleaning 1546 -th movie
Cleaning 1547 -th movie
Cleaning 1548 -th movie
Cleaning 1549 -th movie
Cleaning 1550 -th movie
Cleaning 1551 -th movie
Cleaning 1552 -th movie
Cleaning 1553 -th movie
Cleaning 1554 -th movie
Cleaning 1555 -th movie
Cleaning 1556 -th movie
Cleaning 1557 -th movie
Cleaning 1558 -th movie
Cleaning 1559 -th movie
Cleaning 1560 -th movie
Cleaning 1561 -th movie
Cleaning 1562 -th movie
Cleaning 1563 -th movie
Cleaning 1564 -th movie
Cleaning 1565 -th movie
Cleaning 1566 -th movie
Cleaning 1567 -th movie
Cleaning 1568 -th movie
Cleaning 1569 -th movie
Cleaning 1570 -th movie
Cleaning 1571 -th movie
Cleaning 1572 -th movie
Cleaning 1573 -th movie
Cleaning 1574 -th movie
Cleaning 1575 -t

Cleaning 2367 -th movie
Cleaning 2368 -th movie
Cleaning 2369 -th movie
Cleaning 2370 -th movie
Cleaning 2371 -th movie
Cleaning 2372 -th movie
Cleaning 2373 -th movie
Cleaning 2374 -th movie
Cleaning 2375 -th movie
Cleaning 2376 -th movie
Cleaning 2377 -th movie
Cleaning 2378 -th movie
Cleaning 2379 -th movie
Cleaning 2380 -th movie
Cleaning 2381 -th movie
Cleaning 2382 -th movie
Cleaning 2383 -th movie
Cleaning 2384 -th movie
Cleaning 2385 -th movie
Cleaning 2386 -th movie
Cleaning 2387 -th movie
Cleaning 2388 -th movie
Cleaning 2389 -th movie
Cleaning 2390 -th movie
Cleaning 2391 -th movie
Cleaning 2392 -th movie
Cleaning 2393 -th movie
Cleaning 2394 -th movie
Cleaning 2395 -th movie
Cleaning 2396 -th movie
Cleaning 2397 -th movie
Cleaning 2398 -th movie
Cleaning 2399 -th movie
Cleaning 2400 -th movie
Cleaning 2401 -th movie
Cleaning 2402 -th movie
Cleaning 2403 -th movie
Cleaning 2404 -th movie
Cleaning 2405 -th movie
Cleaning 2406 -th movie
Cleaning 2407 -th movie
Cleaning 2408 -t

Cleaning 3200 -th movie
Cleaning 3201 -th movie
Cleaning 3202 -th movie
Cleaning 3203 -th movie
Cleaning 3204 -th movie
Cleaning 3205 -th movie
Cleaning 3206 -th movie
Cleaning 3207 -th movie
Cleaning 3208 -th movie
Cleaning 3209 -th movie
Cleaning 3210 -th movie
Cleaning 3211 -th movie
Cleaning 3212 -th movie
Cleaning 3213 -th movie
Cleaning 3214 -th movie
Cleaning 3215 -th movie
Cleaning 3216 -th movie
Cleaning 3217 -th movie
Cleaning 3218 -th movie
Cleaning 3219 -th movie
Cleaning 3220 -th movie
Cleaning 3221 -th movie
Cleaning 3222 -th movie
Cleaning 3223 -th movie
Cleaning 3224 -th movie
Cleaning 3225 -th movie
Cleaning 3226 -th movie
Cleaning 3227 -th movie
Cleaning 3228 -th movie
Cleaning 3229 -th movie
Cleaning 3230 -th movie
Cleaning 3231 -th movie
Cleaning 3232 -th movie
Cleaning 3233 -th movie
Cleaning 3234 -th movie
Cleaning 3235 -th movie
Cleaning 3236 -th movie
Cleaning 3237 -th movie
Cleaning 3238 -th movie
Cleaning 3239 -th movie
Cleaning 3240 -th movie
Cleaning 3241 -t

Cleaning 3884 -th movie
Cleaning 3885 -th movie
Cleaning 3886 -th movie
Cleaning 3887 -th movie
Cleaning 3888 -th movie
Cleaning 3889 -th movie
Cleaning 3890 -th movie
Cleaning 3891 -th movie
Cleaning 3892 -th movie
Cleaning 3893 -th movie
Cleaning 3894 -th movie
Cleaning 3895 -th movie
Cleaning 3896 -th movie
Cleaning 3897 -th movie
Cleaning 3898 -th movie
Cleaning 3899 -th movie
Cleaning 3900 -th movie
Cleaning 3901 -th movie
Cleaning 3902 -th movie
Cleaning 3903 -th movie
Cleaning 3904 -th movie
Cleaning 3905 -th movie
Cleaning 3906 -th movie
Cleaning 3907 -th movie
Cleaning 3908 -th movie
Cleaning 3909 -th movie
Cleaning 3910 -th movie
Cleaning 3911 -th movie
Cleaning 3912 -th movie
Cleaning 3913 -th movie
Cleaning 3914 -th movie
Cleaning 3915 -th movie
Cleaning 3916 -th movie
Cleaning 3917 -th movie
Cleaning 3918 -th movie
Cleaning 3919 -th movie
Cleaning 3920 -th movie
Cleaning 3921 -th movie
Cleaning 3922 -th movie
Cleaning 3923 -th movie
Cleaning 3924 -th movie
Cleaning 3925 -t

Cleaning 4616 -th movie
Cleaning 4617 -th movie
Cleaning 4618 -th movie
Cleaning 4619 -th movie
Cleaning 4620 -th movie
Cleaning 4621 -th movie
Cleaning 4622 -th movie
Cleaning 4623 -th movie
Cleaning 4624 -th movie
Cleaning 4625 -th movie
Cleaning 4626 -th movie
Cleaning 4627 -th movie
Cleaning 4628 -th movie
Cleaning 4629 -th movie
Cleaning 4630 -th movie
Cleaning 4631 -th movie
Cleaning 4632 -th movie
Cleaning 4633 -th movie
Cleaning 4634 -th movie
Cleaning 4635 -th movie
Cleaning 4636 -th movie
Cleaning 4637 -th movie
Cleaning 4638 -th movie
Cleaning 4639 -th movie
Cleaning 4640 -th movie
Cleaning 4641 -th movie
Cleaning 4642 -th movie
Cleaning 4643 -th movie
Cleaning 4644 -th movie
Cleaning 4645 -th movie
Cleaning 4646 -th movie
Cleaning 4647 -th movie
Cleaning 4648 -th movie
Cleaning 4649 -th movie
Cleaning 4650 -th movie
Cleaning 4651 -th movie
Cleaning 4652 -th movie
Cleaning 4653 -th movie
Cleaning 4654 -th movie
Cleaning 4655 -th movie
Cleaning 4656 -th movie
Cleaning 4657 -t

Cleaning 5365 -th movie
Cleaning 5366 -th movie
Cleaning 5367 -th movie
Cleaning 5368 -th movie
Cleaning 5369 -th movie
Cleaning 5370 -th movie
Cleaning 5371 -th movie
Cleaning 5372 -th movie
Cleaning 5373 -th movie
Cleaning 5374 -th movie
Cleaning 5375 -th movie
Cleaning 5376 -th movie
Cleaning 5377 -th movie
Cleaning 5378 -th movie
Cleaning 5379 -th movie
Cleaning 5380 -th movie
Cleaning 5381 -th movie
Cleaning 5382 -th movie
Cleaning 5383 -th movie
Cleaning 5384 -th movie
Cleaning 5385 -th movie
Cleaning 5386 -th movie
Cleaning 5387 -th movie
Cleaning 5388 -th movie
Cleaning 5389 -th movie
Cleaning 5390 -th movie
Cleaning 5391 -th movie
Cleaning 5392 -th movie
Cleaning 5393 -th movie
Cleaning 5394 -th movie
Cleaning 5395 -th movie
Cleaning 5396 -th movie
Cleaning 5397 -th movie
Cleaning 5398 -th movie
Cleaning 5399 -th movie
Cleaning 5400 -th movie
Cleaning 5401 -th movie
Cleaning 5402 -th movie
Cleaning 5403 -th movie
Cleaning 5404 -th movie
Cleaning 5405 -th movie
Cleaning 5406 -t

Cleaning 6113 -th movie
Cleaning 6114 -th movie
Cleaning 6115 -th movie
Cleaning 6116 -th movie
Cleaning 6117 -th movie
Cleaning 6118 -th movie
Cleaning 6119 -th movie
Cleaning 6120 -th movie
Cleaning 6121 -th movie
Cleaning 6122 -th movie
Cleaning 6123 -th movie
Cleaning 6124 -th movie
Cleaning 6125 -th movie
Cleaning 6126 -th movie
Cleaning 6127 -th movie
Cleaning 6128 -th movie
Cleaning 6129 -th movie
Cleaning 6130 -th movie
Cleaning 6131 -th movie
Cleaning 6132 -th movie
Cleaning 6133 -th movie
Cleaning 6134 -th movie
Cleaning 6135 -th movie
Cleaning 6136 -th movie
Cleaning 6137 -th movie
Cleaning 6138 -th movie
Cleaning 6139 -th movie
Cleaning 6140 -th movie
Cleaning 6141 -th movie
Cleaning 6142 -th movie
Cleaning 6143 -th movie
Cleaning 6144 -th movie
Cleaning 6145 -th movie
Cleaning 6146 -th movie
Cleaning 6147 -th movie
Cleaning 6148 -th movie
Cleaning 6149 -th movie
Cleaning 6150 -th movie
Cleaning 6151 -th movie
Cleaning 6152 -th movie
Cleaning 6153 -th movie
Cleaning 6154 -t

Cleaning 6845 -th movie
Cleaning 6846 -th movie
Cleaning 6847 -th movie
Cleaning 6848 -th movie
Cleaning 6849 -th movie
Cleaning 6850 -th movie
Cleaning 6851 -th movie
Cleaning 6852 -th movie
Cleaning 6853 -th movie
Cleaning 6854 -th movie
Cleaning 6855 -th movie
Cleaning 6856 -th movie
Cleaning 6857 -th movie
Cleaning 6858 -th movie
Cleaning 6859 -th movie
Cleaning 6860 -th movie
Cleaning 6861 -th movie
Cleaning 6862 -th movie
Cleaning 6863 -th movie
Cleaning 6864 -th movie
Cleaning 6865 -th movie
Cleaning 6866 -th movie
Cleaning 6867 -th movie
Cleaning 6868 -th movie
Cleaning 6869 -th movie
Cleaning 6870 -th movie
Cleaning 6871 -th movie
Cleaning 6872 -th movie
Cleaning 6873 -th movie
Cleaning 6874 -th movie
Cleaning 6875 -th movie
Cleaning 6876 -th movie
Cleaning 6877 -th movie
Cleaning 6878 -th movie
Cleaning 6879 -th movie
Cleaning 6880 -th movie
Cleaning 6881 -th movie
Cleaning 6882 -th movie
Cleaning 6883 -th movie
Cleaning 6884 -th movie
Cleaning 6885 -th movie
Cleaning 6886 -t

Cleaning 7492 -th movie
Cleaning 7493 -th movie
Cleaning 7494 -th movie
Cleaning 7495 -th movie
Cleaning 7496 -th movie
Cleaning 7497 -th movie
Cleaning 7498 -th movie
Cleaning 7499 -th movie
Cleaning 7500 -th movie
Cleaning 7501 -th movie
Cleaning 7502 -th movie
Cleaning 7503 -th movie
Cleaning 7504 -th movie
Cleaning 7505 -th movie
Cleaning 7506 -th movie
Cleaning 7507 -th movie
Cleaning 7508 -th movie
Cleaning 7509 -th movie
Cleaning 7510 -th movie
Cleaning 7511 -th movie
Cleaning 7512 -th movie
Cleaning 7513 -th movie
Cleaning 7514 -th movie
Cleaning 7515 -th movie
Cleaning 7516 -th movie
Cleaning 7517 -th movie
Cleaning 7518 -th movie
Cleaning 7519 -th movie
Cleaning 7520 -th movie
Cleaning 7521 -th movie
Cleaning 7522 -th movie
Cleaning 7523 -th movie
Cleaning 7524 -th movie
Cleaning 7525 -th movie
Cleaning 7526 -th movie
Cleaning 7527 -th movie
Cleaning 7528 -th movie
Cleaning 7529 -th movie
Cleaning 7530 -th movie
Cleaning 7531 -th movie
Cleaning 7532 -th movie
Cleaning 7533 -t

Cleaning 8300 -th movie
Cleaning 8301 -th movie
Cleaning 8302 -th movie
Cleaning 8303 -th movie
Cleaning 8304 -th movie
Cleaning 8305 -th movie
Cleaning 8306 -th movie
Cleaning 8307 -th movie
Cleaning 8308 -th movie
Cleaning 8309 -th movie
Cleaning 8310 -th movie
Cleaning 8311 -th movie
Cleaning 8312 -th movie
Cleaning 8313 -th movie
Cleaning 8314 -th movie
Cleaning 8315 -th movie
Cleaning 8316 -th movie
Cleaning 8317 -th movie
Cleaning 8318 -th movie
Cleaning 8319 -th movie
Cleaning 8320 -th movie
Cleaning 8321 -th movie
Cleaning 8322 -th movie
Cleaning 8323 -th movie
Cleaning 8324 -th movie
Cleaning 8325 -th movie
Cleaning 8326 -th movie
Cleaning 8327 -th movie
Cleaning 8328 -th movie
Cleaning 8329 -th movie
Cleaning 8330 -th movie
Cleaning 8331 -th movie
Cleaning 8332 -th movie
Cleaning 8333 -th movie
Cleaning 8334 -th movie
Cleaning 8335 -th movie
Cleaning 8336 -th movie
Cleaning 8337 -th movie
Cleaning 8338 -th movie
Cleaning 8339 -th movie
Cleaning 8340 -th movie
Cleaning 8341 -t

Cleaning 9151 -th movie
Cleaning 9152 -th movie
Cleaning 9153 -th movie
Cleaning 9154 -th movie
Cleaning 9155 -th movie
Cleaning 9156 -th movie
Cleaning 9157 -th movie
Cleaning 9158 -th movie
Cleaning 9159 -th movie
Cleaning 9160 -th movie
Cleaning 9161 -th movie
Cleaning 9162 -th movie
Cleaning 9163 -th movie
Cleaning 9164 -th movie
Cleaning 9165 -th movie
Cleaning 9166 -th movie
Cleaning 9167 -th movie
Cleaning 9168 -th movie
Cleaning 9169 -th movie
Cleaning 9170 -th movie
Cleaning 9171 -th movie
Cleaning 9172 -th movie
Cleaning 9173 -th movie
Cleaning 9174 -th movie
Cleaning 9175 -th movie
Cleaning 9176 -th movie
Cleaning 9177 -th movie
Cleaning 9178 -th movie
Cleaning 9179 -th movie
Cleaning 9180 -th movie
Cleaning 9181 -th movie
Cleaning 9182 -th movie
Cleaning 9183 -th movie
Cleaning 9184 -th movie
Cleaning 9185 -th movie
Cleaning 9186 -th movie
Cleaning 9187 -th movie
Cleaning 9188 -th movie
Cleaning 9189 -th movie
Cleaning 9190 -th movie
Cleaning 9191 -th movie
Cleaning 9192 -t

Cleaning 9882 -th movie
Cleaning 9883 -th movie
Cleaning 9884 -th movie
Cleaning 9885 -th movie
Cleaning 9886 -th movie
Cleaning 9887 -th movie
Cleaning 9888 -th movie
Cleaning 9889 -th movie
Cleaning 9890 -th movie
Cleaning 9891 -th movie
Cleaning 9892 -th movie
Cleaning 9893 -th movie
Cleaning 9894 -th movie
Cleaning 9895 -th movie
Cleaning 9896 -th movie
Cleaning 9897 -th movie
Cleaning 9898 -th movie
Cleaning 9899 -th movie
Cleaning 9900 -th movie
Cleaning 9901 -th movie
Cleaning 9902 -th movie
Cleaning 9903 -th movie
Cleaning 9904 -th movie
Cleaning 9905 -th movie
Cleaning 9906 -th movie
Cleaning 9907 -th movie
Cleaning 9908 -th movie
Cleaning 9909 -th movie
Cleaning 9910 -th movie
Cleaning 9911 -th movie
Cleaning 9912 -th movie
Cleaning 9913 -th movie
Cleaning 9914 -th movie
Cleaning 9915 -th movie
Cleaning 9916 -th movie
Cleaning 9917 -th movie
Cleaning 9918 -th movie
Cleaning 9919 -th movie
Cleaning 9920 -th movie
Cleaning 9921 -th movie
Cleaning 9922 -th movie
Cleaning 9923 -t

In [32]:
df = pd.DataFrame(clean_data_NC17_addtional)
df['age_rating'].astype("category")
print("The shape of the dataFrame is" , df.shape)
df.head()

The shape of the dataFrame is (9411, 7)


id  runtime                                           genres  \
0  527774    107.0  [Animation, Adventure, Fantasy, Family, Action]   
1  587807    101.0   [Action, Comedy, Family, Animation, Adventure]   
2  484718    110.0                                         [Comedy]   
3  458576    104.0                     [Fantasy, Action, Adventure]   
4  464052    151.0                     [Fantasy, Action, Adventure]   

                                            keywords  \
0  [kung fu, trip, sword, warrior woman, computer...   
1  [computer animation, racial stereotype, live a...   
2  [africa, prince, heir to the throne, sequel, q...   
3  [female soldier, another dimension, bow and ar...   
4  [dc comics, superhero, based on comic, 1980s, ...   

                                production_companies age_rating     budget  
0  [Walt Disney Pictures, Walt Disney Animation S...         PG          0  
1  [Warner Animation Group, Warner Bros. Pictures...         PG   50000000  
2  [Paramount, New Republic Pictures, Eddie Murph...      PG-13   60000000  
3  [CAPCOM, Constantin Film, Impact Pictures, Ten...      PG-13   60000000  
4  [DC Entertainment, Warner Bros. Pictures, The ...      PG-13  200000000

In [33]:
# Extract data that is only in the 5 interest Age_rating (based on the cell below bwlo about certifications)
df = df [ df['age_rating'].isin(['PG','PG-13','R','NC-17','G'])]
print("The shape of the dataFrame is" , df.shape)
df

The shape of the dataFrame is (6959, 7)


id  runtime                                           genres  \
0     527774    107.0  [Animation, Adventure, Fantasy, Family, Action]   
1     587807    101.0   [Action, Comedy, Family, Animation, Adventure]   
2     484718    110.0                                         [Comedy]   
3     458576    104.0                     [Fantasy, Action, Adventure]   
4     464052    151.0                     [Fantasy, Action, Adventure]   
...      ...      ...                                              ...   
9406   28656     90.0                                  [Drama, Horror]   
9407   45042     82.0                                    [Documentary]   
9408   45043    112.0                                          [Drama]   
9409   45047     85.0                                          [Drama]   
9410  389113    110.0                                          [Drama]   

                                               keywords  \
0     [kung fu, trip, sword, warrior woman, computer...   
1     [computer animation, racial stereotype, live a...   
2     [africa, prince, heir to the throne, sequel, q...   
3     [female soldier, another dimension, bow and ar...   
4     [dc comics, superhero, based on comic, 1980s, ...   
...                                                 ...   
9406                     [prison warden, sexploitation]   
9407  [porn star, behind the scenes, porn industry, ...   
9408                     [age difference, erotic movie]   
9409                                     [erotic movie]   
9410                   [prostitute, seoul, south korea]   

                                   production_companies age_rating     budget  
0     [Walt Disney Pictures, Walt Disney Animation S...         PG          0  
1     [Warner Animation Group, Warner Bros. Pictures...         PG   50000000  
2     [Paramount, New Republic Pictures, Eddie Murph...      PG-13   60000000  
3     [CAPCOM, Constantin Film, Impact Pictures, Ten...      PG-13   60000000  
4     [DC Entertainment, Warner Bros. Pictures, The ...      PG-13  200000000  
...                                                 ...        ...        ...  
9406                              [Cinépix, Elite Film]      NC-17          0  
9407  [Berns Brothers Productions, Cafe Sisters Prod...      NC-17          0  
9408         [Shin Cine Communications, Korea Pictures]      NC-17          0  
9409      [Ejve-Film, Filmmakarna, Unicorn Enterprises]      NC-17          0  
9410  [Korean Academy of Film Arts (KAFA), CGV Artho...      NC-17          0  

[6959 rows x 7 columns]

In [34]:
df.to_csv("movie_data_NC_17_additional.csv", header=True,index=False)